core/exception/fixtures/backtrace_locations_in_main.rb

```diff
@@ -0,0 +1,5 @@
+[:test].map do
+  puts Thread.current.backtrace_locations[1].label
```

Comment:
We could use `backtrace_locations(1, 1)[0]` or `backtrace_locations(1..1)[0]` here to avoid walking the entire stack when we only care about the first frame.

Reply:
Done! I changed both versions in the `backtrace_locations_in_[main/required].rb`, but did not change the versions inside `backtrace_locations_inside_nested_blocks` as the `self.backtrace_locations` method already returns all of it anyway.